### ISSUE TITLE
Attribute splattering for TextEdit, NumericEdit, SelectEdit

### ISSUE THREAD

**Issue by brettwinters** [stsrki/Blazorise#236]
Hi @stsrki 

I noticed that your `Size` property (which actually implements the BS `btn-lg`, etc classes) conflicts with the html  [`size`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text#size) property - which for `<input>` elements this is the number of characters wide the element should render (note that this is different from the `minlength` and `maxlength` properties, which you have implemented the latter.

Using lowercase `size` on your component throws a compile exception. 

Not a huge issue, but I guess I recommend that you either (1) allow lowercase `size` to be used or (2) allow `Size` to flow into the the html size (to be standards compliant) and think about another name for the Bootstrap classes...although I realise this is breaking change, so (3) alternatively rename the html `size` to `Width`. ...or (5) ignore me haha

If you consider this, then it would also be great to extend this `Width` or `Size` property to allow `auto` and/or `fit` to automatically adjust the width to the current content. 

For the **`Select`** then I guess I recommend that follow this js component [bootstrap-select](https://developer.snapappointments.com/bootstrap-select/examples/#styling) where 'auto' would be the widest `option` and 'fit' would be the width of the currently selected `option`...

anyway just some ideas which is good to think about at this early stage...

**Comment by stsrki** [comment]
Actually `size` and `Size` are two different things and they should not be intermixed. Upper-case is the recommended way of naming in .NET and in Blazor so I don't plan to change that.

Sizing is meant to be like this: https://getbootstrap.com/docs/4.3/components/forms/#sizing when defining the `Size` attribute. It should not affect the width directly. Maybe I could add some other attribute that could be used to define width of an element.

One other thing. I may consider to add different `Size` enums for different components, eg. SelectSize, TextSize, ButtonSize etc. because Size is maybe too generic for all of them.

**Comment by brettwinters** [comment]
yes, I know its case sensitive and the blazor nomenclature. 

I was just wondering why, say doing this 

```html
<TextEdit class="myclass" style="font-weight:500;" ></TextEdit>
```

is ok (but not recommended I know):
![image](https://user-images.githubusercontent.com/26474006/64221977-de66c880-cef8-11e9-959a-6a3aee048f84.png)

but doing this : 

```html
 <TextEdit size="20" ></TextEdit>
```
causes this:

![image](https://user-images.githubusercontent.com/26474006/64222019-fb030080-cef8-11e9-886f-c249184f3999.png)


But anyway, `Width` is ok - **I just need a way of applying the html (lowercase) `size`** to a `<input>`. Plus if you could have a `Width="auto"` feature that would be great.

Yeah BS Size has different classes for each element : 

| BS class | sizing class  |
| ------------- | ------------- |
| btn   | btn-lg  |
| form-control  | form-control-lg

but I think what you've done to abstract away from that is good. Otherwise too many different types of enums to use...

**Comment by stsrki** [comment]
Probably the first example with `class` and `style` attribute is working because the Blazor is allowing them. I would expect it to break just like for the `size` attribute.

What I'm trying with Blazorise is to abstract as much as possible from html or css or any other "framework". I could create a replacement for "size" attribute or maybe I could add support for [@attributes](https://devblogs.microsoft.com/aspnet/asp-net-core-and-blazor-updates-in-net-core-3-0-preview-7/) parameter that is introduced in the Blazor preview 7.

**Comment by brettwinters** [comment]
yeah, attribute spattering / pass thru would be good - You can't cover all possible the html properties that your users would want to use. Also please consider a `Width` property with helpers such as 'auto' so that the component resizes to fit the content

**Comment by stsrki** [comment]
Closed with: #378 and #379
